<launch>
  <arg name="mav_name" default="firefly" />
  <arg name="enable_logging" default="false" />
  <arg name="enable_ground_truth" default="true" />
  <arg name="log_file" default="$(arg mav_name)" />

  <include file="$(find gazebo_ros)/launch/empty_world.launch">
    <arg name="world_name" value="$(find nbvplanner)/worlds/test_scenario.world" />
    <!-- <arg name="debug" value="true"/> -->
    <arg name="paused" value="true"/>
    <arg name="gui" value="false"/>
  </include>

  <node name="rviz" pkg="rviz" type="rviz" output="screen"/>

  <node name="nbvPlanner" pkg="nbvplanner" type="nbvPlanner" output="screen">
    <param name="tf_frame" type="string" value="world" />
    <param name="resolution" type="double" value="0.4" />
    <param name="use_history" type="bool" value="true" />
    <param name="sensor_max_range" type="double" value="8.0" />
    <remap from="cam0/camera_info" to="/$(arg mav_name)_1/vi_sensor/left/camera_info"/>
    <remap from="cam1/camera_info" to="/$(arg mav_name)_1/vi_sensor/right/camera_info"/>
    <remap from="pointcloud0" to="/$(arg mav_name)_1/vi_sensor/camera_depth/depth/points"/>
    <remap from="pointcloud1" to="/$(arg mav_name)_2/vi_sensor/camera_depth/depth/points"/>
    <remap from="pointcloud2" to="/$(arg mav_name)_3/vi_sensor/camera_depth/depth/points"/>
    <remap from="pointcloud3" to="/$(arg mav_name)_4/vi_sensor/camera_depth/depth/points"/>
    <remap from="pose0" to="$(arg mav_name)_1/odometry_sensor1/pose"/>
    <remap from="nbvplanner0" to="$(arg mav_name)_1/nbvplanner"/>
    <remap from="pose1" to="$(arg mav_name)_2/odometry_sensor1/pose"/>
    <remap from="nbvplanner1" to="$(arg mav_name)_2/nbvplanner"/>
    <remap from="pose2" to="$(arg mav_name)_3/odometry_sensor1/pose"/>
    <remap from="nbvplanner2" to="$(arg mav_name)_3/nbvplanner"/>
    <remap from="pose3" to="$(arg mav_name)_4/odometry_sensor1/pose"/>
    <remap from="nbvplanner3" to="$(arg mav_name)_4/nbvplanner"/>
    <rosparam command="load" file="$(find nbvplanner)/resource/history.yaml" />
  </node>

  <include file="$(find nbvplanner)/launch/mav_inspector.launch">
    <arg name="mav_name" value="firefly" />
    <arg name="subscript" value="1" />
    <arg name="waypoint_x" value="5.0" />
    <arg name="waypoint_y" value="0.0" />
    <arg name="waypoint_z" value="1.0" />
    <arg name="enable_logging" value="$(arg enable_logging)" />
    <arg name="enable_ground_truth" value="$(arg enable_ground_truth)" />
    <arg name="log_file" value="$(arg log_file)" />
  </include>

  <include file="$(find nbvplanner)/launch/mav_inspector.launch">
    <arg name="mav_name" value="firefly" />
    <arg name="subscript" value="2" />
    <arg name="waypoint_x" value="0.0" />
    <arg name="waypoint_y" value="0.0" />
    <arg name="waypoint_z" value="1.0" />
    <arg name="enable_logging" value="$(arg enable_logging)" />
    <arg name="enable_ground_truth" value="$(arg enable_ground_truth)" />
    <arg name="log_file" value="$(arg log_file)" />
  </include>

  <include file="$(find nbvplanner)/launch/mav_inspector.launch">
    <arg name="mav_name" value="firefly" />
    <arg name="subscript" value="3" />
    <arg name="waypoint_x" value="5.0" />
    <arg name="waypoint_y" value="7.0" />
    <arg name="waypoint_z" value="1.0" />
    <arg name="enable_logging" value="$(arg enable_logging)" />
    <arg name="enable_ground_truth" value="$(arg enable_ground_truth)" />
    <arg name="log_file" value="$(arg log_file)" />
  </include>

  <include file="$(find nbvplanner)/launch/mav_inspector.launch">
    <arg name="mav_name" value="firefly" />
    <arg name="subscript" value="4" />
    <arg name="waypoint_x" value="0.0" />
    <arg name="waypoint_y" value="7.0" />
    <arg name="waypoint_z" value="1.0" />
    <arg name="enable_logging" value="$(arg enable_logging)" />
    <arg name="enable_ground_truth" value="$(arg enable_ground_truth)" />
    <arg name="log_file" value="$(arg log_file)" />
  </include>
  
</launch>
